borgmatic/commands/arguments.py

```diff
@@ -865,18 +952,21 @@ def make_parsers():
     )
     borg_group.add_argument('-h', '--help', action='help', help='Show this help message and exit')
 
-    return top_level_parser, subparsers
+    return top_level_parser, subparsers, config_subparsers
```

Comment:
I was having trouble merging them into one, could you tell me how to merge two Subparser objects? I tried merging their choices but that didn't work.

Reply:
I honestly don't know.. I was imagining you'd do something like your existing `choices` merging trick and then stuff that value into a new instance. If that doesn't work out, a couple of different ideas:

 * Instead of returning `subparsers` and `config_subparsers` separately, return them in a tuple or list. The idea is that it could contain any number of subparsers and the calling code would have to handle that.
 * Or, replace `subparsers` and `config_subparsers` with a single return value of subparser _choices_, which you already know how to merge. Although looking at some of the shell completion code that calls `make_parsers()`, there is one case where it also accesses `subparsers._actions` from that return value. So that would have to be accounted for with this approach.

Reply:
I think I found a way to make it work with the choices merging:

```python

    merged_subparsers = argparse._SubParsersAction(None, None, metavar=None, dest='merged', parser_class=None)

    for name, subparser in subparsers.choices.items():
        merged_subparsers._name_parser_map[name] = subparser
        subparser._name_parser_map = merged_subparsers._name_parser_map

    for name, subparser in config_subparsers.choices.items():
        merged_subparsers._name_parser_map[name] = subparser
        subparser._name_parser_map = merged_subparsers._name_parser_map

    return top_level_parser, merged_subparsers        


def parse_arguments(*unparsed_arguments):
    '''
    Given command-line arguments with which this script was invoked, parse the arguments and return
    them as a dict mapping from subparser name (or "global") to an argparse.Namespace instance.
    '''
    top_level_parser, subparsers = make_parsers()
```

This works, and I guess it wouldn't harm other functions calling this too?

Reply:
Nice. It *probably* won't harm other callers, but the way to find out is to try it! For instance, with this code in place, you could generate and source the bash completion and make sure that its tab completion still works as expected, especially when tab-completing action names.